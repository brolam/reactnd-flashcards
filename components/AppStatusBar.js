import React from 'react'
import { StyleSheet } from 'react-native'
import { View, StatusBar } from 'react-native'
import { Constants } from 'expo'
import { red } from '../styles/colors'

export default function AppStatusBar() {
  return (
    <View style={styles.container}>
      <StatusBar style={styles.statusBar} barStyle="light-content" />
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    height: Constants.statusBarHeight,
    backgroundColor: red,
  },
  statusBar: {
    backgroundColor: red,
  }
});