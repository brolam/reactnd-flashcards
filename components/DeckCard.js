import React from 'react'
import { StyleSheet } from 'react-native'
import { View, Text, Platform } from 'react-native'
import PropTypes from 'prop-types'
import { white, gray } from '../styles/colors'
import { width } from 'window-size';

export default function DeckCard({ title, amountOfCards }) {
  return (
    <View style={[
      styles.cardBase,
      Platform.OS === 'ios' ? styles.cardIOS : styles.cardAndroid
    ]}>
      <Text style={styles.title}>{title}</Text>
      <Text style={styles.amountOfCards}>{amountOfCards}</Text>
    </View>
  )
}

DeckCard.propTypes = {
  title: PropTypes.string.isRequired,
  amountOfCards: PropTypes.number.isRequired,
}

const styles = StyleSheet.create({
  cardBase: {
    flexDirection: 'row',
    flex: 1,
    backgroundColor: white,
    padding: 20,
    marginLeft: 10,
    marginRight: 10,
    marginTop: 8,
    shadowOpacity: 0.8,
    shadowColor: 'rgba(0, 0, 0, 0.24)',
  },
  cardIOS: {
    borderRadius: 8,
    shadowRadius: 3,
    shadowOffset: {
      width: 0,
      height: 3
    },
  },
  cardAndroid: {
    borderRadius: 4,
    shadowRadius: 2,
    borderWidth: 1,
    shadowOffset: {
      width: 0,
      height: 2
    },
    elevation: 2,
    borderColor: '#ddd',
  },
  title: { fontSize: 24, textAlign: 'left'},
  amountOfCards: { fontSize: 18, textAlign: 'right' }
});


/*
const styles = StyleSheet.create({
  
})*/