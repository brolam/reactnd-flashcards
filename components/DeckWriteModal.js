import React from 'react'
import { StyleSheet } from 'react-native'
import {
  Modal,
  View,
  Text,
  TextInput,
  TouchableOpacity,
  KeyboardAvoidingView,
  Platform
} from 'react-native'
import CardStyle from '../styles/cardStyles'
import { white, green, red } from '../styles/colors'

export default function DeckWriteModal(props) {
  return (
    <Modal animationType='fade' transparent={true} >
      {Platform.OS === 'ios' ?
        <KeyboardAvoidingView behavior='padding' style={styles.container}>
          <DeckWriteBody {...props} />
        </KeyboardAvoidingView>
        :
        <View style={styles.container} >
          <DeckWriteBody {...props} />
        </View>
      }
    </Modal>
  )
}

function DeckWriteBody(props) {
  return (
    <View style={[CardStyle, styles.body]}>
      <Text style={styles.title}>{props.title}</Text>
      <TextInput
        placeholder="Enter with a Deck Title."
        style={styles.titleTextInput}
        onChange={() => { }} />
      <View style={styles.buttonsContainer}>
        <TouchableOpacity
          style={[buttonStyle, styles.buttonSave]}
          onPress={() => console.log('Save Deck')}>
          <Text style={styles.textButtons}>Save</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={[buttonStyle, styles.buttonCancel]}
          onPress={() => console.log('Cancel Add Deck')}>
          <Text style={styles.textButtons}>Cancel</Text>
        </TouchableOpacity>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(0,0,0,0.5)',
  },
  body: {
    flex: 0,
    flexDirection: 'column',
    alignItems: 'center',
    width: (Platform.OS !== 'ios' ? '80%' : undefined)
  },
  title: {
    fontSize: 18,
    marginBottom: 10,
    fontWeight: '600',
  },
  titleTextInput: {
    fontSize: 24,
    marginBottom: 20,
    height: 50,
    width: '100%'
  },
  buttonsContainer: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    flexWrap: 'wrap',
    marginBottom: 5
  },
  buttonSave: {
    backgroundColor: green,
    margin: 5
  },
  buttonCancel: {
    backgroundColor: red,
    margin: 5
  },
  textButtons: {
    color: white,
    fontSize: 24
  },
});