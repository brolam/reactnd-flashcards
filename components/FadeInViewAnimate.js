import React from 'react';
import { Animated } from 'react-native';

export default class FadeInViewAnimate extends React.Component {
  state = {
    fadeAnim: new Animated.Value(0),  // Initial value for opacity: 0
  }

  componentDidMount() {
    Animated.timing(                  // Animate over time
      this.state.fadeAnim,            // The animated value to drive
      {
        toValue: 3,                   // Animate to opacity: 1 (opaque)
        duration: 8000,              // Make it take a while
      }
    ).start();                        // Starts the animation
  }

  render() {
    let { fadeAnim } = this.state;
    return (
      <Animated.View                 // Special animatable View
        style={{
          ...this.props.style,
          opacity: fadeAnim,         // Bind opacity to animated value
          flexDirection: 'row',
          flex: 1,

        }}
      >
        {this.props.children}
      </Animated.View>
    );
  }
}
