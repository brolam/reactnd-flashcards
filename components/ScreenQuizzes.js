import React from 'react'
import { StyleSheet } from 'react-native'
import { View } from 'react-native'
import PropTypes from 'prop-types'
import QuizCardQuestion from './QuizCardQuestion'


export default function ScreenQuizzes({ quizzes }) {
  return (
    <View style={styles.container}>
      <QuizCardQuestion {...quizzes[0]} />
    </View>
  )
}

ScreenQuizzes.propTypes = {
  quizzes: PropTypes.array.isRequired,
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    flex: 1,
  },
});