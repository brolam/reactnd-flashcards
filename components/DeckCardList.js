import React from 'react'
import { View, FlatList, TouchableOpacity } from 'react-native'
import PropTypes from 'prop-types'
import DeckCard from './DeckCard'

export default function DeckCardList({ decks, onSelectedOneDeck }) {
  return (
    <FlatList
      data={decks}
      renderItem={({ item }) => <DeckCard
        {...item}
        onPress={() => onSelectedOneDeck(item)}
      />}
    />
  )
}

DeckCardList.propTypes = {
  decks: PropTypes.array.isRequired,
}