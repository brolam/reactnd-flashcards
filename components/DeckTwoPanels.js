import React from 'react'
import { StyleSheet } from 'react-native'
import { View, FlatList } from 'react-native'
import PropTypes from 'prop-types'
import DeckCardList from './DeckCardList'

export default function DeckTwoPanel({ decks }) {
  return (
    <View style={styles.container} >
      <View style={styles.containerDeckList}>
        <DeckCardList decks={decks} />
      </View>
    </View>
  )
}

DeckTwoPanel.propTypes = {
  decks: PropTypes.array.isRequired,
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    flex: 1,
  },
  containerDeckList: {
    width: '40%'
  }
})