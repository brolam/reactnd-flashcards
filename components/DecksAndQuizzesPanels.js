import React from 'react'
import { StyleSheet } from 'react-native'
import { View, FlatList } from 'react-native'
import PropTypes from 'prop-types'
import DeckCardList from './DeckCardList'

export default function DecksAndQuizzesPanels({ decks }) {
  return (
    <View style={styles.container} >
      <View style={styles.containerDeckList}>
        <DeckCardList decks={decks} />
      </View>
    </View>
  )
}

DecksAndQuizzesPanels.propTypes = {
  decks: PropTypes.array.isRequired,
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    flex: 1,
  },
  containerDeckList: {
    width: '40%'
  }
})