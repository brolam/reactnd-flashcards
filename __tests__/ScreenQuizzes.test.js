import React from 'react'
import { ScreenQuizzes } from '../components/ScreenQuizzes'

test('renders without crashing', () => {
  const quizzesDummy = [
    { question: 'One Quetion' },
    { question: 'Two Quetion' },
    { question: 'Three Quetion' }
  ]
  const wrapper = shallow(<ScreenQuizzes quizzes = {quizzesDummy} />);
  expect(wrapper).toMatchSnapshot();
});