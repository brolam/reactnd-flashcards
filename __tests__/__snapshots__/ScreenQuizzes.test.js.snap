// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
    }
  }
>
  <QuizCardQuestion
    question="One Quetion"
  />
</View>
`;
